server/container/guice/mailbox/src/main/java/org/apache/james/modules/mailbox/PreDeletionHookModule.java

```diff
@@ -0,0 +1,51 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.modules.mailbox;
+
+import java.util.Set;
+
+import org.apache.commons.configuration.ConfigurationException;
+import org.apache.james.mailbox.extension.PreDeletionHook;
+import org.apache.james.server.core.configuration.ConfigurationProvider;
+
+import com.github.fge.lambdas.Throwing;
+import com.github.steveash.guavate.Guavate;
+import com.google.inject.AbstractModule;
+import com.google.inject.Provides;
+
+public class PreDeletionHookModule extends AbstractModule {
+    @Override
+    protected void configure() {
+        bind(PreDeletionHookLoader.class).to(PreDeletionHookLoaderImpl.class);
+    }
+
+    @Provides
+    PreDeletionHooksConfiguration providesConfiguration(ConfigurationProvider configurationProvider) throws ConfigurationException {
+        return PreDeletionHooksConfiguration.from(configurationProvider.getConfiguration("listeners"));
```

Comment:
Hook will be config on the same with mailbox listener?

Can we need add doc or template for hook configuration?

Reply:
Please contribute Guice bindings separately

Reply:
Hmm I must admit  I did not really wanted to introduce 'yet another' configuration file hence I decided to do that as part of listeners.xml...

Let's do the DOC part a bit later, as documenting the all feature is needed.